src/test/java/net/testudobank/tests/MvcControllerIntegTest.java

```diff
@@ -239,6 +239,69 @@ public void testWithdrawTriggersOverdraft() throws SQLException, ScriptException
   }
 
+/**
+   * Verifies the case where a customer withdraws an amount that causes the
+   * customer to enter overdraft, and the overdraft balance with interest applied
+   * doesn't exceed the overdraft limit.
+   * 
+   * The customer will be given an initial balance of $100 and will withdraw $1099.
+   * This will test the scenario where the overdraft balance itself, which is $100 - $1099 = -$999,
+   * is a valid overdraft balance, but once the 2% interest rate is applied, will cross the limit
+   * of $1000. 
+   * 
+   * Checks to make sure that the customer's balance stays the same as before due to a failed 
+   * withdraw request, and checks that the TransactionHistory table is empty.
+   * 
+   * @throws SQLException
+   * @throws ScriptException
+   */
+
+  @Test
+  public void testWithdrawOverdraftOverage() throws SQLException, ScriptException { 
+
+   //initialize customer1 with a balance of $100. this will be represented as pennies in DB.
+   double CUSTOMER1_BALANCE = 100;
+   int CUSTOMER1_BALANCE_IN_PENNIES = MvcControllerIntegTestHelpers.convertDollarsToPennies(CUSTOMER1_BALANCE);
+   MvcControllerIntegTestHelpers.addCustomerToDB(dbDelegate, CUSTOMER1_ID, CUSTOMER1_PASSWORD, CUSTOMER1_FIRST_NAME, CUSTOMER1_LAST_NAME, CUSTOMER1_BALANCE_IN_PENNIES);
+
+   //Prepare Withdraw Form to withdraw $1099 from this customer's account.
+   double CUSTOMER1_AMOUNT_TO_WITHDRAW = 1099; 
+   User customer1WithdrawFormInputs = new User();
+   customer1WithdrawFormInputs.setUsername(CUSTOMER1_ID);
+   customer1WithdrawFormInputs.setPassword(CUSTOMER1_PASSWORD);
+   customer1WithdrawFormInputs.setAmountToWithdraw(CUSTOMER1_AMOUNT_TO_WITHDRAW);
+
+   //Store the timestamp of the withdraw request to verify it in the TransactionHistory table later
+   LocalDateTime timeWhenWithdrawRequestSent = MvcControllerIntegTestHelpers.fetchCurrentTimeAsLocalDateTimeNoMilliseconds();
+   System.out.println("Timestamp when withdraw request sent: " + timeWhenWithdrawRequestSent);
+
+
+   //verify that customer1's Overdraft balance is equal to remaining withdraw amount with interest applied
+   int CUSTOMER1_ORIGINAL_BALANCE_IN_PENNIES = MvcControllerIntegTestHelpers.convertDollarsToPennies(CUSTOMER1_BALANCE);
+   int CUSTOMER1_AMOUNT_TO_WITHDRAW_IN_PENNIES = MvcControllerIntegTestHelpers.convertDollarsToPennies(CUSTOMER1_AMOUNT_TO_WITHDRAW);
+   int CUSTOMER1_EXPECTED_OVERDRAFT_BALANCE_BEFORE_INTEREST_IN_PENNIES = CUSTOMER1_AMOUNT_TO_WITHDRAW_IN_PENNIES - CUSTOMER1_ORIGINAL_BALANCE_IN_PENNIES;
+   int CUSTOMER1_EXPECTED_OVERDRAFT_BALANCE_AFTER_INTEREST_IN_PENNIES = (int)(CUSTOMER1_EXPECTED_OVERDRAFT_BALANCE_BEFORE_INTEREST_IN_PENNIES * MvcController.INTEREST_RATE);
+   System.out.println("Expected Overdraft Balance in pennies: " + CUSTOMER1_EXPECTED_OVERDRAFT_BALANCE_AFTER_INTEREST_IN_PENNIES);
```

Comment:
You should come up with a better variable name than `CUSTOMER1_EXPECTED_OVERDRAFT_BALANCE` because that variable name is misleading. The actual expected overdraft balance after this transaction is $0 since we expect the withdraw to be rejected. You can try something like `CUSTOMER1_OVERDRAFT_INCREASE_AMOUNT` or something, just don't use the word `"EXPECTED"`.

Reply:
Sure, I'll change that variable name right now to something that is more accurate.